src/k8s-engine/k8s.ts

```diff
@@ -188,31 +226,47 @@ export interface KubernetesBackend extends events.EventEmitter {
   setIntegration(name: string, state: boolean): Promise<string | undefined>;
 
   // #region Events
-
-  /**
-   * Emitted when there has been a change in the progress in the current action.
-   */
-  on(event: 'progress', listener: (progress: { current: number, max: number }) => void): this;
-
-  /**
-   * Emitted when the set of Kubernetes services has changed.
-   */
-  on(event: 'service-changed', listener: (services: ServiceEntry[]) => void): this;
-
-  /**
-   * Emitted when the state of the Kubernetes backend has changed.
-   */
-  on(event: 'state-changed', listener: (state: State) => void): this;
-
-  /**
-   * Emitted when the versions of Kubernetes available has changed.
-   */
-  on(event: 'versions-updated', listener: () => void): this;
-
-  /**
-   * Emitted when k8s is running on a new port
-   */
-  on(event: 'current-port-changed', listener: (port: number) => void): this;
+  addListener<eventName extends keyof KubernetesBackendEvents>(
```

Comment:
Can you give some background on what the this section is doing? I see that the old definitions above got moved higher up into the `KubernetesBackendEvents` interface. What's the next section for? Where would I find documentation on it?

Reply:
This is the same sort of thing we do in [electron-ipc.d.ts](https://github.com/rancher-sandbox/rancher-desktop/blob/33dd5880bdc4d12c33975b506e9baa1152cee83a/src/typings/electron-ipc.d.ts); I'll add comments.

Reply:
Comment added in 71fe56cc5c1da9d02d809f8d6d9462ae89d7ee73; please let me know if we need more details, thanks!

Reply:
nice